### PR TITLE
fix(InputOutput): address issue appending processor id for new intel …

### PR DESCRIPTION
**Pull request by jdhughes-usgs**
…version

* update PARALLEL.md markdown with updated build configurations and instructions
* add Hovenweep build script
* update Denali build script to automatically install in the correct location
* update HPC build instructions
* update main meson.build to build parallel MODFLOW with MPICH on non-cray systems (workaround for meson)

### CODE REVIEW

**Comment by jdhughes-usgs** [comment]
@wpbonelli I have modified the denali build script to directly install in the correct location on Denali based on the version. I used some of the code in `update-module.slurm.batch` to define the installation location. I have not modified `update-module.slurm.batch` but this probably needs to be done. Let me know if there is anything I should modify. I have added you as a reviewer.

**Comment by wpbonelli** [comment]
@jdhughes-usgs I think `update-module.slurm.batch` needs to make configurable the modulefile to use as a template for new versions, it is hardcoded 6.4.2 right now and I guess hovenweep only has 6.5.0.dev0.

But since we now build in the module app dir the step to copy binaries can be removed I think. So e.g.

```shell
#!/bin/bash

#SBATCH --job-name=update-mf6-module
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --time=00:05:00
#SBATCH --output=slurm-%j.out

# this script expects environment variables...
# - MF6_PREV_VERSION, mf6 modulefile version to use as a template
# - MF6_PROJ_ROOT, the path to the modflow6 project root
# - MODULES_PATH, the base path of the module system
MF6_BINDIR="$MF6_PROJ_ROOT/bin"
MF6_LOCAL_VERSION=$(cat "$MF6_PROJ_ROOT/version.txt")

# ...and assumes the module system is laid out as follows
MF6_MODULEFILE_PATH="$MODULES_PATH/modulefiles/modflow/$MF6_LOCAL_VERSION"
MF6_MODULES_PATH="$MODULES_PATH/apps/modflow"
MF6_MODULE_PATH="$MF6_MODULES_PATH/$MF6_LOCAL_VERSION"

# create mf6 modulefile if needed
if [ ! -d "$MF6_MODULE_PATH" ]; then
  rsync "$MODULES_PATH/modulefiles/modflow/$MF6_PREV_VERSION" "$MF6_MODULEFILE_PATH"
  sed -i -e "s/$MF6_PREV_VERSION/$MF6_LOCAL_VERSION/g" "$MF6_MODULEFILE_PATH"
  echo "Created module file: $MF6_MODULEFILE_PATH"
fi

```

The `.gitlab-ci.yml` config also needs an update, looking into that now

File: meson.build [hunk 1]
```diff
@@ -155,9 +155,14 @@ if is_parallel_build
 
   # find mpi
   if is_mpich
-    mpi = dependency('mpich', required : true)
-    mpifort = dependency(mpifort_name, required : mpi.found())
-    dependencies += [ mpi, mpifort ]
+    if is_cray
+      mpi = dependency('mpich', required : true)
+      mpifort = dependency(mpifort_name, required : mpi.found())
+      dependencies += [ mpi, mpifort ]
```
Comment (mjr-deltares) [inline meson.build:161]: Hi @jdhughes-usgs, I was just wondering if and why that mpi dependency on Cray is actually needed

**Comment by jdhughes-usgs** [comment]
@wpbonelli let me know if I missed any of your changes

**Review by wpbonelli — APPROVED** [review]
Looks good, when merged I can update the gitlab mirror and check all is well with the deployment.

**Pull request merged**